### PR TITLE
CLOUDP-278387: Add index for dbUser by project and improve connection secret generation

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Review by s-urbaniak — APPROVED** [review]
looks fine and dandy 👏

File: pkg/indexer/atlasdatabaseuserprojects.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	if user.Spec.Project != nil && user.Spec.Project.Name != "" {
+		project := &akov2.AtlasProject{}
+		err := a.client.Get(a.ctx, *user.Spec.Project.GetObject(user.Namespace), project)
+		if err != nil {
+			a.logger.Errorf("unable to find project to index: %s", err)
+
+			return nil
+		}
+
+		return []string{project.ID()}
```
Comment (s-urbaniak) [inline pkg/indexer/atlasdatabaseuserprojects.go:58]: nit: this can be an empty string silently.

Comment (s-urbaniak) [reply pkg/indexer/atlasdatabaseuserprojects.go:58]: if it is empty string, we shouldn't index, just return nil.

**Review by josvazg — COMMENTED** [review]
Did I miss the reconciler setup in which the indexer is wired for production use?

**Comment by helderjs** [comment]
> Did I miss the reconciler setup in which the indexer is wired for production use?

I think you are mixing up goals. We usually create index for watching purpose, in that case we setup the watcher to use the index to look up the watched object, however it is not the case here.
We are just indexing it to be able to quickly find db users that need connection secret to be generated

**Pull request merged**